src/function.rs

```diff
@@ -288,6 +288,28 @@ where
         self.maybe_changed_after(db, input, revision, cycle_heads)
     }
 
+    fn minimum_serialized_edges(&self, zalsa: &Zalsa, edge: QueryEdge) -> Vec<QueryEdge> {
```

Comment:
Should we change this to a collector pattern to avoid allocating that many `Vec`s?
```suggestion
    fn minimum_serialized_edges(&self, zalsa: &Zalsa, edge: QueryEdge, output: &mut Vec<QueryEdge>) {
```